api/tests/test_authentication.py

```diff
@@ -296,3 +297,27 @@ def test_session_endpoint(client):
 
     assert response.status_code == 200, f"session failed: {response.json}"
     assert 'token' in response.json, 'session succeeded but token field missing from response'
+
+def test_user_signup_rollback(app):
+    """ Verify that a failed signup with cognito 
+    reverts the local DB entry of the user's email."""
+
+
+    rollback_email = 'test_user_signup_rollback@fake.com'
+    signup_response = app.test_client().post(
+        '/api/auth/signup/host',
+        json = {
+            'email': rollback_email,
+            'password': 'lol'
+        }
+    )
+    assert signup_response.status_code == 400
```

Comment:
Due to a recent change in `main` that was merged into this branch, the database does not store a user because the test does not provide a first name. Meaning, this test is no longer testing the given scenario.